client/CServerHandler.cpp

```diff
@@ -764,13 +757,14 @@ void CServerHandler::startCampaignScenario(HighScoreParameter param, std::shared
 				GH.windows().createAndPushWindow<CHighScoreInputScreen>(true, *highScoreCalc);
 			}
 		};
+
+		threadRunLocalServer.join();
```

Comment:
One that you've reported about campaigns? No. 
But it should be fixed in develop, either in this PR or in my PR few days ago.

In any case, this line is gone in #3609 since threadRunLocalServer no longer exists